### ISSUE TITLE
Fix false positives for baseIndentLevel in indentation

### ISSUE THREAD

**Issue by MarkSky** [stylelint/stylelint#4666]
> Clearly describe the bug

When I fix style error on svg tag in html file, I got a strange error:
"Expected indentation of 4 spaces (indentation)"

My html file is:
```
<div>
    <svg xmlns="http://www.w3.org/2000/svg"
             viewBox="0 0 30.8 30.8">
        <defs>
            <style>
                .cls-1,
                .cls-2 {
                    fill: none;
                    stroke: #f66;
                    stroke-miterlimit: 10;
                }

                .cls-1 {
                    stroke-width: 1.5px;
                }

                .cls-2 {
                    stroke-width: 1.8px;
                }

            </style>
        </defs>
        <title>aboutme</title>
        <g id="圖層_2"
              data-name="圖層 2">
            <g id="圖層_1-2"
                  data-name="圖層 1">
                <g id="Group_39"
                      data-name="Group 39">
                    <circle id="Ellipse_1"
                               data-name="Ellipse 1"
                               class="cls-1"
                               cx="15.36"
                               cy="10.83"
                               r="5.37" />
                    <path id="Path_12"
                               data-name="Path 12"
                               class="cls-1"
                               d="M7.57,27.36V24.51a7.83,7.83,0,0,1,7.79-7.88h0a7.83,7.83,0,0,1,7.78,7.88h0v2.85" />
                    <circle id="Ellipse_2"
                               data-name="Ellipse 2"
                               class="cls-2"
                               cx="15.4"
                               cy="15.4"
                               r="14.5" />
                </g>
            </g>
        </g>
    </svg>
</div>
```

If I use `npx stylelint **/*.{html,vue,css,sass,scss} --fix`,
it formats to:
```
<div>
    <svg xmlns="http://www.w3.org/2000/svg"
             viewBox="0 0 30.8 30.8">
        <defs>
            <style>
    .cls-1,
    .cls-2 {
        fill: none;
        stroke: #f66;
        stroke-miterlimit: 10;
    }

    .cls-1 {
        stroke-width: 1.5px;
    }

    .cls-2 {
        stroke-width: 1.8px;
    }

            </style>
        </defs>
        <title>aboutme</title>
        <g id="圖層_2"
              data-name="圖層 2">
            <g id="圖層_1-2"
                  data-name="圖層 1">
                <g id="Group_39"
                      data-name="Group 39">
                    <circle id="Ellipse_1"
                               data-name="Ellipse 1"
                               class="cls-1"
                               cx="15.36"
                               cy="10.83"
                               r="5.37" />
                     <path id="Path_12"
                               data-name="Path 12"
                               class="cls-1"
                                d="M7.57,27.36V24.51a7.83,7.83,0,0,1,7.79-7.88h0a7.83,7.83,0,0,1,7.78,7.88h0v2.85" />
                     <circle id="Ellipse_2"
                                data-name="Ellipse 2"
                                class="cls-2"
                                cx="15.4"
                                cy="15.4"
                                r="14.5" />
                </g>
            </g>
        </g>
    </svg>
</div>
```

In my .stylelintrc file:
```
"indentation": [
    4,
    {
        "baseIndentLevel": 1,
        "ignore": [
            "value"
        ],
        "indentClosingBrace": false
    }
],
```

I think the rule should let css be indented 1 level than <style> tag, but in fact, the css is indented 1 level from root.

**Comment by jeddy3** [comment]
@MarkSky Thanks for the report and for using the template.

I can reproduce this [using the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIA8UCWA3AfAHQHYAJD8kBnAFwE8AbGHAohgQ32DwfaMiogCd58eMKAG42HfAF8xhJAHpy1WnjloseEABoQJCrjKMAHghAALMgFsqm8BFwAzVAHNjrXNhA8ArjRLv+rEFRcWD1GMlRbP3wAbWl8ABYNONdxfHcAI0YSGABJYJg9ABkYdBgqKIBGJPoOdydcXhgo2JrUtJB0RipPJpA4hgBdatS6-L0AYW4SIMcAIR5GMF7+Oy7suKlcAbwJCRAJIA).

> I think the rule should let css be indented 1 level than <style> tag, but in fact, the css is indented 1 level from root.

Yes, that's the [intended behaviour of the rule](https://stylelint.io/user-guide/rules/indentation#baseindentlevel-intauto).

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by MarkSky** [comment]
Thank you for your help. 
I'm these weeks, but I will to read the Developer guide.

**Comment by jeddy3** [comment]
Closing [as stylistic rules are frozen](https://github.com/stylelint/stylelint/issues/5674). The community is welcome to migrate the rule [to a plugin](https://stylelint.io/developer-guide/plugins) and fix this bug.

**Comment by nolimitdev** [comment]
I think this should be reopened because of it is a bug. See doc here https://stylelint.io/user-guide/rules/indentation/#optional-secondary-options 

> By default, the indent level of the CSS code block in non-CSS-like files is determined by the shortest indent of non-empty line. The setting baseIndentLevel allows you to define a relative indent level based on CSS code block opening or closing line.
> 
> For example, with [ 2, { baseIndentLevel: 1 } ], CSS should be indented 1 levels higher than <style> tag:

```
<html lang="en">
  <head>
    <style>
      a {
        display: block;
      }
    </style>
  </head>
</html>
```

So `baseIndentLevel` by doc must start from `<style>` tag level not from document/root level. So this is bug because relative indent does not work as documented.

**Comment by jeddy3** [comment]
@nolimitdev We [deprecated the stylistic rules in `15.0.0`](https://stylelint.io/migration-guide/to-15#deprecated-stylistic-rules), including `indentation`, and we'll be removing them in the next major release.

We now recommend using a pretty-printer (like Prettier) alongside Stylelint. Linters and pretty printers are complementary tools that work together to help you write consistent and error-free code.

Use Stylelint to:

- [avoid errors](https://stylelint.io/user-guide/rules#avoid-errors)
- [enforce (non-stylistic conventions)](https://stylelint.io/user-guide/rules#enforce-conventions)

Use a pretty printer to format stylistic things like whitespace to an optimal line length.

Alternatively, you can [write a Stylelint plugin](https://stylelint.io/developer-guide/plugins) to enforce whatever whitespace conventions you want.

**Comment by nolimitdev** [comment]
@jeddy3 thanks for explanation, I did not notice that bugs wont fixed.  Btw 76 of the rules that enforce stylistic conventions are deprecated because of Prettier exists. But Prettier is opinionated with minimum configuration so how many of that 76 stylelint rules will be configurable via Prettier?

**Comment by jeddy3** [comment]
>  But Prettier is opinionated with minimum con,figuration so how many of that 76 stylelint rules will be configurable via Prettier? Prettier does not seem to be alternative to deprecated rules.

Making the rules configurable came at the cost of unsustainable complexity. Prettier avoids this by being more opintioned. Our recommendation is for people to adopt their conventions for whitespace and then focus on [customising](https://stylelint.io/user-guide/customize) Stylelint's [built-in rules for enforcing (non-stylistic) consistency](https://stylelint.io/user-guide/rules#enforce-conventions) to bring their CSS under control.

Having said that, you're welcome to continue using Stylelint to enforce whitespace conventions by [migrating any of the built-in rules you need to a plugin](https://stylelint.io/developer-guide/plugins). We won't support them because of the aforementioned cost, but people can take that on if they wish.
